src/env.cpp

```diff
@@ -163,17 +163,18 @@ struct action_t {
     argtokey_callback_t freeKey;
 };
 
-int deleteValue; // pointer to this as the value represents a delete
+thread_local static int deleteValue; // pointer to this as the value represents a delete
```

Comment:
I wouldn't think we would actually need to change deleteValue for threads, it is merely being used a static pointer location (to differentiate between NULL), there is never any value assigned or changed. There is probably a much more correct way of doing this (having a reserved constant pointer that won't conflict with other pointers). All the other static variables do need to change though.

Reply:
if it's read-only, makes sense